### PR TITLE
Added --input_format_csv_trim_whitespaces option

### LINKED ISSUE #22574: CSV parsing setting: format_csv_not_trim_spaces
**Use case**
CSV file with unquoted strings and spaces at start and end of strings which needs to be preserved.

>  In violation of the RFC, when parsing rows without quotes, the leading and trailing spaces and tabs are ignored. 

```
echo "aa ,2" | clickhouse-local -S "a String, b int" --input-format="CustomSeparated" --format_custom_escaping_rule=CSV --format_custom_field_delimiter=',' --format_csv_delimiter=',' -q "select * from table FORMAT CSV"
"aa",2
```

**Describe the solution you'd like**
Setting

https://github.com/ClickHouse/ClickHouse/issues/6742#issuecomment-526567265

### PR DESCRIPTION
**Pull request by Demilivor**
Closes #22574

### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added --input_format_csv_trim_whitespaces option

### Description
Trims spaces and tabs in non-quoted CSV strings.

Default value: `true`.

**Examples**

Query

```bash
echo '  string  ' | ./clickhouse local -q  "select * from table FORMAT CSV" --input-format="CSV" --input_format_csv_trim_whitespaces=true
```

Result

```text
"string"
```

Query

```bash
echo '  string  ' | ./clickhouse local -q  "select * from table FORMAT CSV" --input-format="CSV" --input_format_csv_trim_whitespaces=false
```

Result

```text
"  string  "
```

### CODE REVIEW

**Comment by rvasin** [comment]
@rschu1ze Please be reviewer of this PR and assign 'can be tested' label.

**Comment by rschu1ze** [comment]
> @rschu1ze Please be reviewer of this PR and assign 'can be tested' label.

@Avogar: since you know the import/export code much better than I do - would like to be assignee for this PR?

File: src/Processors/Formats/Impl/CSVRowInputFormat.cpp [hunk 1]
```diff
@@ -280,7 +282,16 @@ bool CSVFormatReader::readField(
...
+    if (format_settings.csv.trim_whitespaces) [[likely]]
+        skipWhitespacesAndTabs(*buf);
+    else if (type->isNullable())
+    {
+        auto nested_type = typeid_cast<const DataTypeNullable *>(type.get())->getNestedType();
+        if (!isStringOrFixedString(nested_type))
+            skipWhitespacesAndTabs(*buf);
+    }
+    else if (!isStringOrFixedString(type))
+        skipWhitespacesAndTabs(*buf);
```
Comment (Avogar) [inline src/Processors/Formats/Impl/CSVRowInputFormat.cpp:294]:
```suggestion
    if (format_settings.csv.trim_whitespaces || !isStringOrFixedString(RemoveNullable(type))) [[likely]]
        skipWhitespacesAndTabs(*buf);
```

Comment (Demilivor) [reply src/Processors/Formats/Impl/CSVRowInputFormat.cpp]: Done.
Thanks for suggestion!

**Comment by Avogar** [comment]
Hi! No more comments, it's ready for merge. But we have some issues with our CI (with build checks) that don't allow to merge PRs

**Pull request merged**